File: plugins/vehicle_plugins/src/bicycle.cpp
#include "vehicle_plugins/bicycle.hpp"
#include "pluginlib/class_list_macros.hpp"

namespace vehicle_plugins
{

void Bicycle::move()
{
    RCLCPP_INFO(rclcpp::get_logger("Bicycle"), "Bicycle is moving smoothly ! 🚲");
}

void Bicycle::stop()
{
    RCLCPP_INFO(rclcpp::get_logger("Bicycle"), "Bicycle stopped !");
}

std::string Bicycle::getInfo()
{
    return "Bicycle - 2 wheels - eco friendly vehicle";
}

}  // namespace vehicle_plugins

PLUGINLIB_EXPORT_CLASS(vehicle_plugins::Bicycle, vehicle_base::VehicleBase)

File: ros2_fundamentals/plugins/vehicle_plugins_py/vehicle_plugins_py/bicycle.py
from vehicle_base_py.vehicle_base import VehicleBase

class Bicycle(VehicleBase):
    def move(self) -> None:
        print("Bicycle is moving smoothly ! 🚲")

    def stop(self) -> None:
        print("Bicycle stopped !")

    def get_info(self) -> str:
        return "Bicycle - 2 wheels - eco friendly vehicle"